mypy/types.py

```diff
@@ -188,6 +188,15 @@ def __init__(self,
     def accept(self, visitor: 'TypeVisitor[T]') -> T:
         return visitor.visit_unbound_type(self)
 
+    def __hash__(self) -> int:
```

Comment:
I haven't reviewed this file but I can do it later (even after the PR has been landed) since this looks pretty straightforward.